In [1]:
import pandas as pd

In [2]:
DATADIR = "/cellar/users/snwright/Data/RareCommon/AZdata/"

## Updated quant

In [38]:
quant = pd.read_csv(DATADIR+"Table17_pan_quant.txt", sep="\t", usecols=['pheno','phenotype_dedup' ,'geno', 'p'])

In [39]:
quant_df = quant.sort_values('p').drop_duplicates(subset=['pheno', 'geno', 'phenotype_dedup'])

In [40]:
quant_df = quant_df.loc[quant_df.p< 1e-6, :]

In [44]:
pheno_counts = quant_df.pheno.value_counts()
keep_phenos = pheno_counts[pheno_counts>5].index

20015 - sitting_height
20151 - forced vital capacity
23098 - weight
3063 - FEV1
5133 - strong meridian
30680 - calcium

In [52]:
keep_phenos

Int64Index([   50, 20015, 30100, 30050, 30040, 30250, 30070, 30270,    51,
            30240, 30630, 23129, 30260, 30760, 30290, 23130, 30080, 30300,
            23101, 30750, 30010, 30880, 30090, 23102, 30110, 30660, 30840,
            30120, 30690, 23125, 30700, 30770, 30180, 23105, 23110, 30030,
            23113, 23118, 23121, 30870, 30200, 23117, 23114, 30000, 23122,
            30020, 30780, 20153, 23126, 30640, 30130, 30730, 23106, 30610,
            30060, 23109, 30280,  3062, 30830, 30150, 30140, 23108, 23107,
            30720, 23098, 30210,   102, 30190, 20151,    47, 21002, 30680,
            30600, 30860, 30670, 30890,  3143,  3063, 30620,    48, 20257,
            30650, 20258,  2976, 23120, 23128,  5133, 30740,    46],
           dtype='int64')

## Update binary

In [8]:
binary = pd.read_csv(DATADIR+"Table16_pan_binary.txt", sep="\t", usecols=['phenotype', 'genotype', 'EUR_pValue'])

In [14]:
binary_df = binary.sort_values('EUR_pValue').drop_duplicates(subset=['phenotype', 'genotype'])

In [21]:
binary_df = binary_df.loc[binary_df.EUR_pValue < 1e-6, :]

In [22]:
pheno_counts = binary_df.phenotype.value_counts()
keep_phenos = pheno_counts[pheno_counts>5].index

2453 -cancer diagnosed by doctor
C92 - Diagnoses - C92 Nyeloid leukemia
C81-C96 - primary lymphoid and hematopoietic malignant neoplasms
D46 - myelodusplastic syndromes
Congenital malformations, deformations and chromosomal abnormalities
D47 - Other neoplasms of uncertain


In [25]:
keep_phenos

Index(['40006#ChapterII#Chapter II Neoplasms',
       'Union#ChapterII#Chapter II Neoplasms',
       '41202#ChapterII#Chapter II Neoplasms',
       'Union#E80#E80 Disorders of porphyrin and bilirubin metabolism',
       'Source of report of E80 (disorders of porphyrin and bilirubin metabolism)',
       '40012#3#Malignant| primary site', '2453#Cancer diagnosed by doctor',
       '41202#C92#C92 Myeloid leukaemia',
       '40006#BlockC81-C96#C81-C96 Malignant neoplasms| stated or presumed to be primary| of lymphoid| haematopoietic and related tissue',
       'Union#D46#D46 Myelodysplastic syndromes',
       '40011#2#(8010-8790) Epithelial',
       'Union#ChapterXVII#Chapter XVII Congenital malformations| deformations and chromosomal abnormalities',
       'Union#C92#C92 Myeloid leukaemia',
       'Union#D47#D47 Other neoplasms of uncertain or unknown behaviour of lymphoid| haematopoietic and related tissue',
       'Union#C920#C92.0 Acute myeloid leukaemia',
       'Union#E78#E78 Disorder

In [3]:
binary_phenos = pd.read_csv(DATADIR + "Table8_top_binary.txt.pheno_counts", sep="\t", header=None)
binary_phenos.columns = ["pheno", "count"]
binary_phenos = binary_phenos.sort_values(by="count", ascending=False)

In [15]:
binary_phenos[binary_phenos["count"]>= 5]["pheno"].tolist()

['Union#E80#E80 Disorders of porphyrin and bilirubin metabolism',
 'Source of report of E80 (disorders of porphyrin and bilirubin metabolism)',
 'Union#ChapterXVII#Chapter XVII Congenital malformations| deformations and chromosomal abnormalities',
 '40006#ChapterII#Chapter II Neoplasms',
 '41202#ChapterII#Chapter II Neoplasms',
 'Union#ChapterII#Chapter II Neoplasms',
 '40011#8500#Duct adenocarcinoma',
 'Union#C509#C50.9 Breast| unspecified',
 '2453#Cancer diagnosed by doctor']

## Binary Candidate List
* E80
* 40011 - 8500 ; not in pan ukbb
* C50.9 - not in pan ukbb, only C50

In [16]:
quant_phenos = pd.read_csv(DATADIR + "Table8_top_quant.txt.pheno_counts", sep="\t", header=None)
quant_phenos.columns = ["pheno", "count"]
quant_phenos = quant_phenos.sort_values(by="count", ascending=False)

In [18]:
quant_phenos.head(20)

pheno  count
132                                  Standing height     39
98                      Mean corpuscular haemoglobin     26
100                          Mean corpuscular volume     25
122  Red blood cell (erythrocyte) distribution width     23
101               Mean platelet (thrombocyte) volume     23
103                         Mean sphered cell volume     21
127                                   Sitting height     19
51                                   HDL cholesterol     18
68             High light scatter reticulocyte count     18
69        High light scatter reticulocyte percentage     16
123                               Reticulocyte count     16
135                                  Total bilirubin     16
124                          Reticulocyte percentage     15
37                                  Direct bilirubin     15
50                      Glycated haemoglobin (HbA1c)     15
117                                   Platelet count     15
118                                    Platelet crit     14
16                                  Apolipoprotein A     14
102                         Mean reticulocyte volume     14
143                                            Urate     13

In [17]:
quant_phenos[quant_phenos["count"]>= 5]["pheno"].tolist()

['Standing height',
 'Mean corpuscular haemoglobin',
 'Mean corpuscular volume',
 'Red blood cell (erythrocyte) distribution width',
 'Mean platelet (thrombocyte) volume',
 'Mean sphered cell volume',
 'Sitting height',
 'HDL cholesterol',
 'High light scatter reticulocyte count',
 'High light scatter reticulocyte percentage',
 'Reticulocyte count',
 'Total bilirubin',
 'Reticulocyte percentage',
 'Direct bilirubin',
 'Glycated haemoglobin (HbA1c)',
 'Platelet count',
 'Platelet crit',
 'Apolipoprotein A',
 'Mean reticulocyte volume',
 'Urate',
 'Creatinine',
 'Trunk fat-free mass',
 'Trunk predicted mass',
 'Red blood cell (erythrocyte) count',
 'IGF-1',
 'Whole body water mass',
 'Triglycerides',
 'Whole body fat-free mass',
 'Cholesterol',
 'Platelet distribution width',
 'Basal metabolic rate',
 'Haemoglobin concentration',
 'LDL direct',
 'Apolipoprotein B',
 'Cystatin C',
 'Mean corpuscular haemoglobin concentration',
 'Haematocrit percentage',
 'Gamma glutamyltransferase',
 'Alk

## Quantitative candidates

* 50 - no in pan ukbb
* 30050
* 23406 - not in pan ukbb
* 30250
* 30840
* 30630
* 30640
* 30880
* 23478 - not in pan ukbb
* 30140
* 23110
* 30830